File: src/imager-ts/ImagerFormat.ts
interface FormatListInterface {
    [key: number]: string;
}

const FORMAT_JPG = 1;
const FORMAT_JPEG = 2;
const FORMAT_GIF = 3;
const FORMAT_PNG = 4;
const FORMAT_APNG = 5;
const FORMAT_JPE = 6;
const FORMAT_JIF = 7;
const FORMAT_JFIF = 8;
const FORMAT_JFI = 9;
const FORMAT_WEBP = 10;
const FORMAT_AVIF = 11;
const FORMAT_HEIF = 12;
const FORMAT_HEIC = 13;

export var FormatList: FormatListInterface = {}
FormatList[FORMAT_JPG] = 'jpg'
FormatList[FORMAT_JPEG] = 'jpeg'
FormatList[FORMAT_GIF] = 'gif'
FormatList[FORMAT_PNG] = 'png'
FormatList[FORMAT_APNG] = 'apng'
FormatList[FORMAT_JPE] = 'jpe'
FormatList[FORMAT_JIF] = 'jif'
FormatList[FORMAT_JFIF] = 'jfif'
FormatList[FORMAT_JFI] = 'jfi'
FormatList[FORMAT_WEBP] = 'webp'
FormatList[FORMAT_AVIF] = 'avif'
FormatList[FORMAT_HEIF] = 'heif'
FormatList[FORMAT_HEIC] = 'heic'


export function ImagerFormat(format: string): number | false {
    format = format.toLowerCase();
    for (var k in FormatList) {
        if (format == FormatList[k]) {
            return parseInt(k);
        }
    }
    return false
}

File: src/imager-go/ImagerFormat.go
package imagergo

import "strings"

type format uint8
type formatList map[format]string
type listFormat map[string]format

const (
	formatJpg  format = 1
	formatJpeg format = 2
	formatGif  format = 3
	formatPng  format = 4
	formatApng format = 5
	formatJpe  format = 6
	formatJif  format = 7
	formatJfif format = 8
	formatJfi  format = 9
	formatWebp format = 10
	formatAvif format = 11
	formatHeif format = 12
	formatHeic format = 13
)

var Formats formatList = formatList{
	formatJpg:  "jpg",
	formatJpeg: "jpeg",
	formatGif:  "gif",
	formatPng:  "png",
	formatApng: "apng",
	formatJpe:  "jpe",
	formatJif:  "jif",
	formatJfif: "jfif",
	formatJfi:  "jfi",
	formatWebp: "webp",
	formatAvif: "avif",
	formatHeif: "heif",
	formatHeic: "heic",
}

var local struct {
	list  []string
	lists listFormat
}

func init() {
	if local.lists == nil {
		local.list = []string{}
		local.lists = listFormat{}
		for format, value := range Formats {
			local.list = append(local.list, value)
			local.lists[value] = format
		}
	}
}

// получаем указатель формата
func GetFormat(format string) (res format, is bool) {
	format = strings.ToLower(format)
	res, is = local.lists[format]
	return
}

func FormatsList() []string {
	return local.list
}

func FormatName(f uint8) string {
	return format(f).Name()
}

// получаем формат из указателя
func (format format) Name() string {
	value, is := Formats[format]
	if is {
		return value
	}
	return ""
}
